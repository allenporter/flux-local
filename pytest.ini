[pytest]
log_level = DEBUG
asyncio_mode = auto
