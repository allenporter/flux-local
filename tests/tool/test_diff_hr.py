"""Tests for the flux-local `diff hr` command."""

import pytest

from syrupy.assertion import SnapshotAssertion

from . import run_command


@pytest.mark.parametrize(
    ("args", "env"),
    [
        (["podinfo", "-n", "podinfo", "--path", "tests/testdata/cluster/"], {}),
        (
            [
                "renovate",
                "-n",
                "default",
                "--path",
                "tests/testdata/cluster6/",
                "-a",
                "batch/v1/CronJob",
            ],
            {},
        ),
        (
            [
                "podinfo",
                "-n",
                "podinfo",
                "--path",
                "tests/testdata/cluster/",
                "-o",
                "yaml",
            ],
            {},
        ),
        (["podinfo", "--all-namespaces", "--path", "tests/testdata/cluster/"], {}),
        (["--all-namespaces", "--path", "tests/testdata/cluster/"], {}),
        (
            ["podinfo", "-n", "podinfo", "--path", "tests/testdata/cluster/"],
            {"DIFF": "diff"},
        ),
        (
            [
                "podinfo",
                "-n",
                "podinfo",
                "--path",
                "tests/testdata/cluster/",
                "--limit-bytes",
                "20",
            ],
            {"DIFF": "diff"},
        ),
        (
            [
                "podinfo",
                "-n",
                "podinfo",
                "--path",
                "tests/testdata/cluster/",
                "--limit-bytes",
                "20000",
            ],
            {},
        ),
        (["example", "--path", "tests/testdata/cluster"], {}),
        (["example", "--namespace", "example", "--path", "tests/testdata/cluster"], {}),
        (["example", "--all-namespaces", "--path", "tests/testdata/cluster"], {}),
    ],
    ids=[
        "podinfo",
        "cluster6",
        "yaml",
        "all-namespaces",
        "all",
        "external",
        "external-limit-bytes",
        "limit-bytes",
        "not-found",
        "not-found-namespace",
        "not-found-all-namespaces",
    ],
)
async def test_diff_hr(
    args: list[str], env: dict[str, str], snapshot: SnapshotAssertion
) -> None:
    """Test test diff hr commands."""
    result = await run_command(["diff", "hr"] + args, env=env)
    assert result == snapshot
