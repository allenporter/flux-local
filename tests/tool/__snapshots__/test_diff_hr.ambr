# serializer version: 1
# name: test_diff_hr[all-namespaces]
  ''
# ---
# name: test_diff_hr[all]
  ''
# ---
# name: test_diff_hr[cluster6]
  ''
# ---
# name: test_diff_hr[external-limit-bytes]
  ''
# ---
# name: test_diff_hr[external]
  ''
# ---
# name: test_diff_hr[limit-bytes]
  ''
# ---
# name: test_diff_hr[not-found-all-namespaces]
  '''
  HelmRelease object 'example' not found in 'None' namespace
  
  '''
# ---
# name: test_diff_hr[not-found-namespace]
  '''
  HelmRelease object 'example' not found in 'example' namespace
  
  '''
# ---
# name: test_diff_hr[not-found]
  '''
  HelmRelease object 'example' not found in 'flux-system' namespace
  
  '''
# ---
# name: test_diff_hr[podinfo]
  ''
# ---
# name: test_diff_hr[yaml]
  ''
# ---
